ownCloud/Diagnostic/DiagnosticViewController.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  DiagnosticViewController.swift
+//  ownCloud
+//
+//  Created by Felix Schwarz on 27.07.20.
+//  Copyright © 2020 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2020, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+
+import UIKit
+import ownCloudSDK
+
+protocol DiagnosticNodeGenerator : OCActivity {
+	var isDiagnosticNodeGenerationAvailable : Bool { get }
+	func provideDiagnosticNode(for context: OCDiagnosticContext, completion: @escaping (_ groupNode: OCDiagnosticNode?, _ style: DiagnosticViewController.Style) -> Void)
+}
+
+class DiagnosticViewController: StaticTableViewController {
+
+	var context : OCDiagnosticContext?
+
+	var rootNode : OCDiagnosticNode?
+	var nodes : [OCDiagnosticNode]? {
+		didSet {
+			rebuildTable()
+		}
+	}
+
+	enum Style {
+		case flat
+		case hierarchical
+	}
+
+	init(for node: OCDiagnosticNode, context: OCDiagnosticContext?, style: Style = .hierarchical) {
+		self.context = context
+		self.style = style
+
+		super.init(style: .grouped)
+
+		self.rootNode = node
+
+		self.navigationItem.title = node.label
+		self.navigationItem.rightBarButtonItem = UIBarButtonItem(image: UIImage(named: "open-in"), style: .plain, target: self, action: #selector(self.shareAsMarkdown(_:)))
```

Comment:
Would be nice to have an `accessibilityLabel`, when the `UIBarButtonItem` is image only